### ISSUE TITLE
TOC only works in Mozilla Firefox, but not in Safari or Chrome

### ISSUE THREAD

**Issue by kyra1990** [thaider/Tweeki#86]
TOC is only displayed in Mozilla Firefox.
I have checked if it works on your official website, but it does not work either.
Example: http://tweeki.thai-land.at/wiki/Accordions

Thanks.

**Comment by thaider** [comment]
Thanks for the report. To clarify: Do you mean that the TOC is not shown at all or that the scrollspy feature (highlighting of the current section) is not working?

**Comment by kyra1990** [comment]
The scrollspy feature.
The toclevel-1 is shown but the  toclevel-2 and the others do not.

Thanks.

**Comment by thaider** [comment]
I patched Bootstrap's javascript, because scrollspy didn't work with special characters (we have a lot of them in German). It seems that this patch works only for Firefox and makes scrollspy dysfunctional for all the other browsers. What I did was changing this line in [bootstrap.js](https://github.com/twbs/bootstrap/blob/v3.3.7/dist/js/bootstrap.js#L1946):

```
var $href = /^#./.test(href) && $(href)
```

to this:

```
var $href = /^#./.test(href) && $('[id="'+e.substr(1)+'"]')
```

The problem is, that encodings for special signs contain periods which are interpreted by jQuery as class selectors. So, instead of directly using the href attribute as a selector, I used the `[name="value"]` notation for the `id` attribute where we can have an arbitrary string as an attribute's value. I don't know, why it's not working in Safari & Chrome. Some idea?

**Comment by benvarkey** [comment]
Actually, I tried with the original bootstrap.min.js and the issue is still present. However, from Chrome's developer console, I can do the following to get the TOC working again:
```javascript
$.getScript('https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js');
$.getScript('https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js');
$(this).scrollspy("refresh");
```
However, this kills the top navbar functionality.
Hope this helps.

**Comment by thaider** [comment]
The problem was to be found in my hack to make the fixed navbar work. Headlines appear to need padding to be positioned correctly, my solution was to use empty spans as anchors with padding and a negative margin. However, scrollspy tests for visibility and those empty elements are not considered to be visible (by some browsers, that is), because they don't consume any screen space.

I found a solution to the problem by adding another hack: the span now has a left padding and negative margin of 1px width. There should definitely exist more elegant solutions for this problem but for the moment it seems to work... Would be happy if you could confirm this, see e.g. here http://tweeki.thai-land.at/wiki/Configuration_Options#Left_Sidebar

**Comment by benvarkey** [comment]
TOC works in Chrome with the latest pull. Thanks!

**Comment by kyra1990** [comment]
TOC works in Chrome and Safari. What files have you modified? Thanks!

**Comment by thaider** [comment]
As explained above, I just hacked the styling. So the only file that changed is `components/tweeki-base.less`. See [this commit](https://github.com/thaider/Tweeki/commit/75867c89b3bcf0397e19da65c19d6c8be853e166)…

**Comment by kyra1990** [comment]
It works perfectly. Thank you very much for your help!
